mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+                    max_value = to_plot.max()
+
+                    # Cut top and bottom of data as specified in cut variables
+                    min_value = min_value + (max_value-min_value)*cut_min
+                    max_value = max_value*cut_max
+
+                # Check the size of the array to be plotted
+                # print(to_plot.shape)
+
+                # Set the axis (might be switched?)
+                x_axis_mult = data.plot_options.x_limit_multiplier
+                y_axis_mult = data.plot_options.y_limit_multiplier
+
+                X = np.linspace(data.metadata.limits[0]*x_axis_mult,
+                                data.metadata.limits[1]*x_axis_mult,
+                                data.metadata.dimension[0]+1)
+                Y = np.linspace(data.metadata.limits[2]*y_axis_mult,
+                                data.metadata.limits[3]*y_axis_mult,
+                                data.metadata.dimension[1]+1)
+
+                # Create a meshgrid for both x and y
+                x, y = np.meshgrid(X, Y)
+
+                # Generate information on necessary colorrange
+                levels = MaxNLocator(nbins=150).tick_values(min_value,
+                                                            max_value)
+
+                # Select colormap
+                cmap = plt.get_cmap('hot')
+                norm = BoundaryNorm(levels, ncolors=cmap.N, clip=True)
+
+                # Create the figure
+                fig, (ax0) = plt.subplots()
+
+                # Plot the data on the meshgrids
+                if data.plot_options.log:
+                    color_norm = matplotlib.colors.LogNorm(vmin=min_value,
+                                                           vmax=max_value)
+                    im = ax0.pcolormesh(x, y, to_plot,
+                                        cmap=cmap, norm=color_norm)
+                else:
+                    im = ax0.pcolormesh(x, y, to_plot, cmap=cmap, norm=norm)
+
+                # Add the colorbar
+                fig.colorbar(im, ax=ax0)
+
+                # Add a title
+                ax0.set_title(data.metadata.title)
+                
+                # Add axis labels
+                plt.xlabel(data.metadata.xlabel)
+                plt.ylabel(data.metadata.ylabel)
+
+                if data.plot_options.custom_ylim_top:
+                    ax0.set_ylim(top=data.plot_options.top_lim)
+
+                if data.plot_options.custom_ylim_bottom:
+                    ax0.set_ylim(bottom=data.plot_options.bottom_lim)
+
+                if data.plot_options.custom_xlim_left:
+                    ax0.set_xlim(left=data.plot_options.left_lim)
+
+                if data.plot_options.custom_xlim_right:
+                    ax0.set_xlim(right=data.plot_options.right_lim)
+
+            else:
+                print("Error, dimension not read correctly")
+
+        plt.show()
+
+
+class make_sub_plot:
+    """
+    make_plot plots contents of McStasData objects
```

Comment:
```suggestion
    make_sub_plot plots contents of McStasData objects
```
copied and pasted from above?

Reply:
Added additional text, but make_plot and make_sub_plot are still close in their description, as the functions are rather similar.